tests/testTriggers.go

```diff
@@ -7960,6 +7960,163 @@ func TriggerMetroDRMigrationSchedule(contexts *[]*scheduler.Context, recordChan
 	})
 }
 
+// AggrVolDepReplResizeOps crashes vol driver
+func TriggerAggrVolDepReplResizeOps(contexts *[]*scheduler.Context, recordChan *chan *EventRecord) {
+	defer ginkgo.GinkgoRecover()
+	defer endLongevityTest()
+	startLongevityTest(AggrVolDepReplResizeOps)
+	event := &EventRecord{
+		Event: Event{
+			ID:   GenerateUUID(),
+			Type: AggrVolDepReplResizeOps,
+		},
+		Start:   time.Now().Format(time.RFC1123),
+		Outcome: []error{},
+	}
+
+	defer func() {
+		event.End = time.Now().Format(time.RFC1123)
+		*recordChan <- event
+	}()
+
+	setMetrics(*event)
+
+	stepLog := "Repl factor update and Volume Update on aggregated volume"
+	Step(stepLog, func() {
+		log.InfoD(stepLog)
+		log.Infof("Starting test case here !!")
+		// Validate minimum set of Storage nodes needed to create aggregated Volumes
+		sNodes := node.GetStorageNodes()
+		if len(sNodes) <= 3 {
+			err := fmt.Errorf("Node needs minimum of 3 Nodes to create aggregated volumes")
+			log.FailOnError(err, "is io running on the volume?")
+		}
+
+		aggrApplication := []string{"aggr-mysql", "postgres"}
+		currAppList := Inst().AppList
+
+		revertAppList := func() {
+			Inst().AppList = currAppList
+		}
+		defer revertAppList()
+
+		Inst().AppList = []string{}
+		for _, eachApp := range aggrApplication {
+			Inst().AppList = append(Inst().AppList, eachApp)
+		}
+
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			*contexts = append(*contexts, ScheduleApplications(fmt.Sprintf("aggrvoldeprepresize-%d", i))...)
+		}
+		ValidateApplications(*contexts)
+
+		allVolsCreated := []*volume.Volume{}
+		for _, eachContext := range *contexts {
+			vols, err := Inst().S.GetVolumes(eachContext)
+			if err != nil {
+				log.Errorf("Failed to get app %s's volumes", eachContext.App.Key)
+			}
+			for _, eachVol := range vols {
+				allVolsCreated = append(allVolsCreated, eachVol)
+			}
+		}
+
+		if len(allVolsCreated) < 1 {
+			err := fmt.Errorf("no volumes created with the contexts")
+			log.FailOnError(err, "volume created successfully?")
+		}
+
+		teardownContext := func() {
+			opts := make(map[string]bool)
+			opts[scheduler.OptionsWaitForResourceLeakCleanup] = true
+
+			for _, ctx := range *contexts {
+				TearDownContext(ctx, opts)
+			}
+		}
+		defer teardownContext()
+
+		setReplFactor := func(volName *volume.Volume) {
+			getReplicaSets, err := Inst().V.GetReplicaSets(volName)
+			log.FailOnError(err, "Failed to get replication factor on the volume")
+
+			if len(getReplicaSets[0].Nodes) == 3 || len(getReplicaSets[0].Nodes) == 1 {
+				err := Inst().V.SetReplicationFactor(volName, 2, nil, nil, true)
```

Comment:
Can we add more validation related to aggregated volumes node allocation.can you check the documentation for calculations once and them

Reply:
sure

Reply:
fixed the validation part for aggregated volume